src/com/yetanalytics/lrs_admin_ui/handlers.cljs

```diff
@@ -111,12 +123,29 @@
 (re-frame/reg-event-fx
  :login/success-handler
  global-interceptors
- (fn [{:keys [db]} [_ {:keys [json-web-token]}]]
-   (let [{:keys [username]} (::db/login db)]
-     {:fx [[:dispatch [:session/set-token json-web-token]]
-           [:dispatch [:session/set-username username]]
-           [:dispatch [:login/set-password nil]]
-           [:dispatch [:login/set-username nil]]]})))
+ (fn [_ [_ {:keys [json-web-token]}]]
+   {:fx [[:dispatch [:session/set-token json-web-token]]
+         [:dispatch [:login/set-password nil]]
+         [:dispatch [:login/set-username nil]]]}))
```

Comment:
So you're removing the `:session/set-username` dispatch, not just in proxy JWT mode but also during regular login?

Reply:
I see that it's because `:session/set-token` now calls `:session/get-me`.
- The handlers are declared out of order (like `:session/set-token` comes _after_ `:session/get-me` in the namespace), so it took me a bit to understand the program flow.
- More substantively, now there's an additional HTTP call, even in non-proxy mode. Are you sure we can't short-circuit by keeping the old code in non-proxy mode?

Reply:
- Ordering: I noted that myself and tried a few different configurations. The reality is _some_ will be out of order because of the branching chain of events but I can take another whack at making sense of it
- No, it was actually an irresponsible pattern, and kind of a lazy shorcut, to display user input that way in the first place. It is super common to retrieve user info directly from the server on login, not cache it from the username field on login page. Not only that but the setting of username does not prevent pageload or user flow, it is async because it is in fact the token-setter that triggers the login. The db call is indexed on ID, and deals with extremely little data.

Reply:
@kelvinqian00 this ordering change is the only one needed to where everything is below its invocation. I'm not really sure its super relevant since they're event handlers not functions and things call up all the time in handlers, including in this same file, but maybe it reads easier now. https://github.com/yetanalytics/lrs-admin-ui/pull/47/commits/97885bdb0f0836966635dfa563aec4792350fb6f

Reply:
That makes it better, thanks